### PR TITLE
IOTSTOOL-2162: Slate docs update etc

### PR DESCRIPTION
**Pull request by mark-edgeworth**
* Updates to docs to use foundation examples
* Changed Mbed Cloud references to Pelion Device Management
* Updated copyright messages
* Added check for OpenSSL version

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -49,39 +49,23 @@ documentation <https://cloud.mbed.com/docs/latest/mbed-cloud-sdk-python/>`__:
 1. Create an API key in the `Pelion Device Management
    Portal <https://portal.us-east-1.mbedcloud.com/>`__.
```
Comment (madchutney) [inline README.rst]: This is now: https://portal.mbedcloud.com

File: README.rst [hunk 2]
```diff
@@ -49,39 +49,23 @@ documentation <https://cloud.mbed.com/docs/latest/mbed-cloud-sdk-python/>`__:
 1. Create an API key in the `Pelion Device Management
```
Comment (madchutney) [inline README.rst]: The above section can probably be deleted as they are a bit redundant:

These instructions can also be found in the `official
documentation <https://cloud.mbed.com/docs/latest/mbed-cloud-sdk-python/>`__:

File: README.rst [hunk 3]
```diff
@@ -49,39 +49,23 @@ documentation <https://cloud.mbed.com/docs/latest/mbed-cloud-sdk-python/>`__:
...
-       {
-         "email": "username@example.org",
-         "full_name": "A.N. Individual",
-         ...
-       }
+        # List the first 10 devices on your Pelion Device Management account.
+        for device in Device().list(max_results=10):
+            print("Hello device %s" % device.name)
 
 Documentation and examples
```
Comment (madchutney) [inline README.rst]: These documentation and examples section needs to be reworked and point to the slate documentation.

File: README.rst [hunk 4]
```diff
@@ -49,39 +49,23 @@ documentation <https://cloud.mbed.com/docs/latest/mbed-cloud-sdk-python/>`__:
 1. Create an API key in the `Pelion Device Management
    Portal <https://portal.us-east-1.mbedcloud.com/>`__.
 
-2. Create a configuration file in your ``$HOME`` or project directory
-   (``.mbed_cloud_config.json``):
 
-   .. code:: javascript
+2. Create a configuration file ``.env`` in your ``$HOME`` or project
```
[the remark this answers is not in the document]
Comment (mark-edgeworth) [reply README.rst]: From python-dotenv docs: "...which will load environment variables from a file named .env in the current directory or any of its parents or from the path specificied (sic)". The home directory is included; a brief test on my system indicates that it is included.

File: README.rst (hunk 1 above)
Comment (mark-edgeworth) [reply README.rst]: Fixed, along with similar reference in the SSL section at the bottom.

File: README.rst (hunk 2 above)
Comment (mark-edgeworth) [reply README.rst]: Removed.

**Pull request merged**